Allow MATCHES_STRING for single string

jramosf opened the pull request:
See: https://docs.split.io/reference/matcher-type#matches_string

The `string` (missing attribute) or `strings` depends on the matcher type.

jramosf commented:
@davidji99 any chance this can get reviewed/merged? Thanks!

davidji99 reviewed (changes requested):
Good start but there's quite more work needed. You'll need to:

1. Modify https://github.com/davidji99/terraform-provider-split/blob/master/split/resource_split_split_definition.go#L484-L506 to actually accept the new attribute.
2. Update the API client: https://github.com/davidji99/terraform-provider-split/blob/master/api/split_definition.go#L74
3. Add test for the `split_split_defition` resource
4. Update tests for `split_split_defition`

jramosf commented:
> Good start but there's quite more work needed. You'll need to:
> 
> 1. Modify https://github.com/davidji99/terraform-provider-split/blob/master/split/resource_split_split_definition.go#L484-L506 to actually accept the new attribute.
> 2. Update the API client: https://github.com/davidji99/terraform-provider-split/blob/master/api/split_definition.go#L74
> 3. Add test for the `split_split_defition` resource
> 4. Update tests for `split_split_defition`

Thanks for the review @davidji99 . I've made changes to the Pull Request including checking multiple matchers at the same time, and fixed the initial typo `MATCH_STRING`. Tested with `app.split.io` and via `make test` and everything working ok so far.

<img width="670" alt="image" src="https://github.com/davidji99/terraform-provider-split/assets/9470856/750f75d5-c5d5-4504-940a-7feaf4ea3858">

davidji99 reviewed (approved):
Thanks looks good.

One last thing is you need to update the docs, but I'll do that for you.
